packages/browser/src/client/tester/context.ts

```diff
@@ -122,13 +122,13 @@ export const userEvent: UserEvent = {
   keyboard(text: string) {
     return triggerCommand('__vitest_keyboard', text)
   },
-  hover(element: Element) {
```

Comment:
Yeah, I agree, but atm I'm not in a position to contribute that bit as well, still very new to the vitest's codebase. If that's a blocker then I'd appreciate some assistance on getting this covered.

Reply:
There is a `test/browser/test/userEvent.test.ts` file. You just need to add a test that makes sure the position is correct when hovering the element (using `addEventListener` most likely)

Reply:
There is already test with hover/unhover - I think just checking the cursor position in the even should be enough

Reply:
So what was expected to be a simple test turned out to be not so. After a few hours and experimenting with repo's settings around `ui: false`, `headless` a.s.o, as well as running the suite via regex pattern I arrived at no conclusive result.

Case in point:
![image](https://github.com/user-attachments/assets/54274dd4-3f1f-4e76-ab2d-b08cc37f146b)
![image](https://github.com/user-attachments/assets/9126dd2d-14e0-416a-b66b-11418a312296)
![image](https://github.com/user-attachments/assets/71495700-8fc3-48e4-971b-e0e66425509f)

1. Whilst running the whole file and `ui: true` (default), the results are rather nonsensical, and seem to be static. I think it might have something to do with pointer not being reset correctly between events, and moving into the target element from unexpected angle.
2. Second is more silly, as it's only off by one, and with no browser interface to interfere my only guess is that iframe introduces fractional css height / offset.
3. The last one passes, as it runs only that suite in isolation.

Overall, whilst I appreciate wanting to have this change covered, the bug seems to fall under `strict fix`, so unless there's something obvious that I'm missing to fix the test easily, I'm not convinced that lack thereof should be blocking this bug fix?

P.S. We could create a suite that would spy on the invoked functions, but it'd be rather close to testing the implementation, rather then behaviour, IMO.

Reply:
I think this just proves that we need to spend more time working on how coordinates are reported. Maybe we need to hide the UI while the test is running for example

We can merge this PR if there is a test that checks that at least one option is working correctly except positions - maybe `modifiers`?

Reply:
Yeah I agree regarding the reporting bit and covering / understanding it better, but neither my bandwidth nor understanding of vitest's UI / browser suite is anywhere near enough sufficient to accomplish that. 

I have pushed a change that asserts `shift` and `control` modifier keys. All tests pass locally via `pnpm run test:ci` but I'm not sure why the browser ones in CI are failing. I don't think that I introduced new suites, so not certain I understand why the number of tests ran is off by one against the expected.

Reply:
`test:ci` doesn't run browser tests. `test:browser:webdriverio` and `test:browser:playwright` do

In your case, it seems like you need to limit the test to `playwright` because `webdriverio` doesn't support modifiers

Reply:
Yeah fair, must've been too later, found the suites in `test/browser`. 

Kinda see what you mean regarding redeclaring variables, just based it off the existing test for hover/unhover. I separated it into a standalone, `playwright`-only test (preview also doesn't pass options IIRC, but I also see a few more tests failing with preview, presuming it's intended). 

Not sure why Firefox is unhappy on Windows in CI, lmk if there's something I'm missing. Playwright doesn't seem to have any notes about `options` support in Firefox.

Reply:
The click test is flaky for some reason, I don't think the fail is related to your test